Hadoop incompatible when using ADL UFS

**Alluxio Version:**
Master 2.6.0-SNAPSHOT-ee131efc279e141a151c00f3c201fb934efc7b46

**Describe the bug**
Data is already distributed loaded into Alluxio.
ImageNet training job is listing and reading data through Alluxio Fuse mount point.
```
    -Dalluxio.worker.fuse.enabled=true -Dalluxio.worker.fuse.mount.alluxio.path=/ -Dalluxio.worker.fuse.mount.point=/mnt/alluxio-fuse 
    -Dalluxio.worker.fuse.mount.options=allow_other,kernel_cache,max_read=131072,attr_timeout=7200,entry_timeout=7200 
```

The training job can successfully pass the list directory phrase but will be stuck and eventually timeout in the reading data phrase. The stuck behavior happens sometimes at the beginning of the reading job (happened once) or after reading for about 30 minutes (happened twice). 

After the training job reading data timeout, running `cat` command in one of the workers will hang for a long time and eventually succeed.  The other workers don't have this behavior and are functioning normally. The worker has the hanging `cat` behavior has the following error message in their worker.log:
```
2021-05-05 08:02:25,463 ERROR BlockReadHandler - Failed to run DataReader.
com.google.common.util.concurrent.ExecutionError: java.lang.ExceptionInInitializerError
Caused by: java.lang.ExceptionInInitializerError
	at org.apache.hadoop.fs.azure.AzureNativeFileSystemStore$1.eventOccurred(AzureNativeFileSystemStore.java:2013)
Caused by: java.lang.NullPointerException
	at org.apache.hadoop.util.VersionInfo.<init>(VersionInfo.java:46)
	at org.apache.hadoop.util.VersionInfo.<clinit>(VersionInfo.java:99)
	... 43 more
2021-05-07 20:00:46,499 ERROR BlockReadHandler - Failed to run DataReader.
com.google.common.util.concurrent.ExecutionError: java.lang.NoClassDefFoundError: Could not initialize class org.apache.hadoop.util.VersionInfo
  at alluxio.worker.block.UnderFileSystemBlockReader.updateUnderFileSystemInputStream(UnderFileSystemBlockReader.java:314)
  at alluxio.worker.block.UnderFileSystemBlockReader.init(UnderFileSystemBlockReader.java:138)
Caused by: java.lang.NoClassDefFoundError: Could not initialize class org.apache.hadoop.util.VersionInfo
  at org.apache.hadoop.fs.azure.AzureNativeFileSystemStore$1.eventOccurred(AzureNativeFileSystemStore.java:2013)
  at org.apache.hadoop.fs.FileSystem.open(FileSystem.java:914)
```
Full useful worker log for the hanging `cat` worker [error.log](https://github.com/Alluxio/alluxio/files/6446721/error.log)

For this hanging `cat` worker:
```
    "Worker.BytesReadRemote.10.240.0.14": {
      "count": 0
    },
    "Worker.Fuse.BytesRead.10.240.0.14": {
      "count": 40817501
    },
    "Worker.Fuse.BytesToRead.10.240.0.14": {
      "count": 60370944
    },
```
There is no `BytesReadDirect`, and `BytesReadRemote` is 0, which means that nothing got to read from this worker. 
All other workers find BytesReadDirect and BytesReadRemote
```
    "Worker.BytesReadDirect.10.240.0.10": {
      "count": 274375900
    },
    "Worker.BytesReadRemote.10.240.0.10": {
      "count": 10439234567
    },
```
No deadlock found from the hanging `cat` worker jstack [jstack.log](https://github.com/Alluxio/alluxio/files/6462185/jstack.log)

--- a/shaded/adl/pom.xml
+++ b/shaded/adl/pom.xml
@@ -1,0 +1,179 @@
+<!--
+
+    The Alluxio Open Foundation licenses this work under the Apache License, version 2.0
+    (the "License"). You may not use this work except in compliance with the License, which is
+    available at www.apache.org/licenses/LICENSE-2.0
+
+    This software is distributed on an "AS IS" basis, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND,
+    either express or implied, as more fully set forth in the License.
+
+    See the NOTICE file distributed with this work for information regarding copyright ownership.
+
+-->
+<project xmlns="http://maven.apache.org/POM/4.0.0" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+         xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/xsd/maven-4.0.0.xsd">
+    <modelVersion>4.0.0</modelVersion>
+    <parent>
+        <groupId>org.alluxio</groupId>
+        <artifactId>alluxio-shaded</artifactId>
+        <version>2.6.0-SNAPSHOT</version>
+    </parent>
+    <artifactId>alluxio-shaded-adl</artifactId>
+    <packaging>jar</packaging>
+    <name>Alluxio Shaded Libraries - Adl</name>
+    <description>Shaded Adl Module</description>
+
+    <properties>
+        <build.path>${project.parent.parent.basedir}/build</build.path>
+        <adl.hadoop.version>2.10.0</adl.hadoop.version>
+        <shading.prefix>alluxio.shaded.adl</shading.prefix>
+    </properties>
+
+    <dependencies>
+        <dependency>
+            <groupId>org.apache.logging.log4j</groupId>
+            <artifactId>log4j-api</artifactId>
+            <scope>provided</scope>
+        </dependency>
+        <dependency>
+            <groupId>org.apache.logging.log4j</groupId>
+            <artifactId>log4j-core</artifactId>
+            <scope>provided</scope>
+        </dependency>
+        <dependency>
+            <groupId>org.slf4j</groupId>
+            <artifactId>slf4j-api</artifactId>
+            <scope>provided</scope>
+        </dependency>
+        <dependency>
+            <groupId>org.apache.logging.log4j</groupId>
+            <artifactId>log4j-slf4j-impl</artifactId>
+            <scope>provided</scope>
+        </dependency>
+        <dependency>
+            <groupId>org.apache.hadoop</groupId>
+            <artifactId>hadoop-azure-datalake</artifactId>
+            <version>${adl.hadoop.version}</version>
+        </dependency>
+        <dependency>
+            <groupId>org.apache.hadoop</groupId>
+            <artifactId>hadoop-common</artifactId>
+            <version>${adl.hadoop.version}</version>
+        </dependency>
+        <dependency>
+            <groupId>org.apache.hadoop</groupId>
+            <artifactId>hadoop-hdfs</artifactId>
+            <version>${adl.hadoop.version}</version>
+        </dependency>
+    </dependencies>
+
+    <build>
+        <plugins>
+            <plugin>
+                <groupId>org.apache.maven.plugins</groupId>
+                <artifactId>maven-deploy-plugin</artifactId>
+                <configuration>
+                    <skip>true</skip>
+                </configuration>
+            </plugin>
+            <plugin>
+                <groupId>org.apache.maven.plugins</groupId>
+                <artifactId>maven-shade-plugin</artifactId>
+                <executions>
+                    <execution>
+                        <id>shade-alluxio</id>
+                        <phase>package</phase>
+                        <goals>
+                            <goal>shade</goal>
+                        </goals>
+                        <configuration>
+                            <!-- Since we don't shade all artifacts, we need to promote the transitive dependencies -->
+                            <promoteTransitiveDependencies>true</promoteTransitiveDependencies>
+                            <createDependencyReducedPom>${create.dependency.reduced.pom}</createDependencyReducedPom>
+                            <relocations>
+                                <relocation>
+                                    <pattern>com.google</pattern>
+                                    <shadedPattern>${shading.prefix}.com.google</shadedPattern>
+                                </relocation>
+                                <relocation>
+                                    <pattern>org.objectweb.asm</pattern>
+                                    <shadedPattern>${shading.prefix}.org.objectweb.asm</shadedPattern>
+                                </relocation>
+                                <relocation>
+                                    <pattern>org.jboss.netty</pattern>
+                                    <shadedPattern>${shading.prefix}.org.jboss.netty</shadedPattern>
+                                </relocation>
+                                <relocation>
+                                    <pattern>io.netty</pattern>
+                                    <shadedPattern>${shading.prefix}.io.netty</shadedPattern>
+                                </relocation>
+                                <relocation>
+                                    <pattern>org.apache.curator</pattern>
+                                    <shadedPattern>${shading.prefix}.org.apache.curator</shadedPattern>
+                                </relocation>
+                                <relocation>
+                                    <pattern>org.apache.http</pattern>
+                                    <shadedPattern>${shading.prefix}.org.apache.http</shadedPattern>
+                                </relocation>
+                                <relocation>
+                                    <pattern>org.apache.commons.httpclient</pattern>
+                                    <shadedPattern>${shading.prefix}.org.apache.commons.httpclient</shadedPattern>
+                                </relocation>
+                                <relocation>
+                                    <pattern>org.htrace</pattern>
+                                    <shadedPattern>${shading.prefix}.org.htrace</shadedPattern>
+                                </relocation>
+                                <relocation>
+                                    <pattern>org.codehaus.jackson</pattern>
+                                    <shadedPattern>${shading.prefix}.org.codehaus.jackson</shadedPattern>
+                                </relocation>
+                            </relocations>
+                            <transformers>
+                                <transformer implementation="org.apache.maven.plugins.shade.resource.ServicesResourceTransformer" />
+                            </transformers>
+                            <artifactSet>
+                                <includes>
+                                    <include>com.google.guava:*</include>
+                                    <include>com.google.protobuf:*</include>
+                                    <include>com.google.code.findbugs:*</include>
+                                    <include>asm:asm</include>
+                                    <include>io.netty:netty:*</include>
+                                    <include>io.netty:netty-all:*</include>
+                                    <include>org.apache.curator:*</include>
+                                    <include>org.apache.hadoop:*</include>
+                                    <include>org.htrace:htrace-core</include>
+                                    <include>net.java.dev.jets3t:jets3t</include>
+                                    <include>org.apache.httpcomponents:*</include>
+                                    <include>commons-httpclient:commons-httpclient</include>
+                                    <include>org.codehaus.jackson:*</include>
+                                </includes>
+                                <excludes>
+                                    <exclude>org.slf4j:*</exclude>
+                                    <exclude>log4j:*</exclude>
+                                </excludes>
+                            </artifactSet>
+
+                            <filters>
+                                <!-- Exclude signatures -->
+                                <filter>
+                                    <artifact>*:*</artifact>
+                                    <excludes>
+                                        <exclude>META-INF/*.SF</exclude>
+                                        <exclude>META-INF/*.DSA</exclude>
+                                        <exclude>META-INF/*.RSA</exclude>
+                                    </excludes>
+                                </filter>
+                                <filter>
+                                    <artifact>commons-httpclient:commons-httpclient</artifact>
+                                    <excludes>
+                                        <exclude>META-INF/LICENSE.txt</exclude>
+                                    </excludes>
+                                </filter>
+                            </filters>
+                        </configuration>
+                    </execution>
+                </executions>
+            </plugin>
+        </plugins>
+    </build>
+</project>

--- a/shaded/cosn/pom.xml
+++ b/shaded/cosn/pom.xml
@@ -28,6 +28,7 @@
     <cosn.hadoop-cos.version>3.1.0-5.8.5</cosn.hadoop-cos.version>
     <cosn.cos_api.version>5.6.19</cosn.cos_api.version>
     <cosn.hadoop.version>3.1.0</cosn.hadoop.version>
+    <shading.prefix>alluxio.shaded.cosn</shading.prefix>
   </properties>
 
   <dependencies>

--- a/shaded/ozone/pom.xml
+++ b/shaded/ozone/pom.xml
@@ -27,7 +27,7 @@
     <!-- run properly from sub-project directories -->
     <build.path>${project.parent.parent.basedir}/build</build.path>
     <ufs.ozone.version>1.0.0</ufs.ozone.version>
-    <shading.prefix>alluxio.shaded.hdfs</shading.prefix>
+    <shading.prefix>alluxio.shaded.ozone</shading.prefix>
   </properties>
 
   <dependencies>

--- a/shaded/pom.xml
+++ b/shaded/pom.xml
@@ -23,6 +23,7 @@
   <description>Parent POM for different shaded Libraries</description>
 
   <modules>
+    <module>adl</module>
     <module>client</module>
     <module>cosn</module>
     <module>hadoop</module>

--- a/underfs/adl/pom.xml
+++ b/underfs/adl/pom.xml
@@ -30,24 +30,12 @@
   </properties>
 
   <dependencies>
-    <!-- External dependencies -->
-    <dependency>
-      <groupId>org.apache.hadoop</groupId>
-      <artifactId>hadoop-azure-datalake</artifactId>
-      <version>${adl.hadoop.version}</version>
-    </dependency>
-    <dependency>
-      <groupId>org.apache.hadoop</groupId>
-      <artifactId>hadoop-common</artifactId>
-      <version>${adl.hadoop.version}</version>
-    </dependency>
-    <dependency>
-      <groupId>org.apache.hadoop</groupId>
-      <artifactId>hadoop-hdfs</artifactId>
-      <version>${adl.hadoop.version}</version>
-    </dependency>
-
     <!-- Internal dependencies -->
+    <dependency>
+      <groupId>org.alluxio</groupId>
+      <artifactId>alluxio-shaded-adl</artifactId>
+      <version>${project.version}</version>
+    </dependency>
     <dependency>
       <groupId>org.alluxio</groupId>
       <artifactId>alluxio-core-common</artifactId>

